fetch_data.js

```diff
@@ -0,0 +1,67 @@
+var fs = require('fs');
+var http = require('http');
+
+/**
+ * Function that parses dates from .csv files in ./input.
+ * Passes the most recent date to the callback.
+ * @callback {fetchPublishedData} cb - Fetches latest metadata and checks if it is newer
+ */
+function  checkForNewData(cb){
+ fs.readdir('./input',function(err,files){
+   var newDate
+   var oldDate = new Date(2000, 01, 01)
+   files.forEach(file => {
+     var year = file.slice(0,4)
+     var month = file.slice(4,6)
+     var day = file.slice(6,8)
+     newDate = new Date(year, month, day)
+     if (oldDate < newDate) {
+       oldDate = newDate
+     }
+   })
+   return cb(newDate);
+ });
+}
+
+
+/**
+ * HTTP requests for latest published General Government Employees Demographics dataset.
+ * @param {Date} - The date of the last downloaded data set
+ */
+function fetchPublishedData(lastDownloaded) {
```

Comment:
Instead of checking each file on disk (above), I'd just make this naively try to

* fetch the latest rows
* try and write the rows to disk based on the `rowsUpdatedAt` date

If it fails (because the file already exists) we can just print an error message.  It's dumb, because it does a little unnecessary work in fetching the CSV when it might not have to, but it's dead simple.